packages/concerto-vocabulary/lib/vocabulary.js

```diff
@@ -144,16 +144,47 @@ class Vocabulary {
      * @returns {*} an object with missingTerms and additionalTerms properties
      */
     validate(modelFile) {
-        const getOwnProperties = (d) => {
-            // ensures we have a valid return, even for scalars
-            return d.getOwnProperties?.() ? d.getOwnProperties?.() : [];
+        const getOwnProperties = (declaration) => {
+            // ensures we have a valid return, even for scalars and map-declarations
```

Comment:
Are those the only types of declarations out there?

Reply:
Yes. There is logic below to filter that out.